### PR TITLE
[WIP] Add external TLS-origination tests

### PR DESCRIPTION
**Pull request by vadimeisenbergibm**
It accesses www.cloudflare.com, which has HTTP and HTTPS site versions
the HTTPS version responds with Expect-CT header, this is how it can be
distinguished from the HTTP version.

Expect-CT header: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Expect-CT

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by howardjohn** [comment]
Also there are efforts to move tests to test/integration folder so we can run locally as well, and I believe there are some folks running those tests air gapped so this would not work well there

**Comment by vadimeisenbergibm** [comment]
@howardjohn Yes we have such tests https://github.com/istio/istio/blob/master/tests/e2e/tests/pilot/externalservice_test.go.

Once we decide not to use external sites in tests, we will have to rewrite those tests too.

File: tests/e2e/tests/pilot/externalservice_tls_origination_test.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
+// Copyright 2017 Istio Authors
```
Comment (hzxuzhonghu) [inline tests/e2e/tests/pilot/externalservice_tls_origination_test.go]: 2019

File: tests/e2e/tests/pilot/externalservice_tls_origination_test.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+			config:            "testdata/networking/v1alpha3/tls-origination-cloudflare.yaml",
+			url:               "http://www.cloudflare.com",
+			shouldBeReachable: true,
+			shouldBeTLS:       true,
+		},
+		{
+			name:              "UNREACHABLE_www.google.com_443_over_cloudflare_443",
+			config:            "testdata/networking/v1alpha3/tls-origination-cloudflare.yaml",
+			url:               "https://www.google.com",
+			shouldBeReachable: false,
```
Comment (hzxuzhonghu) [inline tests/e2e/tests/pilot/externalservice_tls_origination_test.go]: This case sounds unreasonable.

File: tests/e2e/tests/pilot/externalservice_tls_origination_test.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+		},
+		{
+			name:              "REACHABLE_WITH_TLS_www.cloudflare.com_443_over_cloudflare_443",
+			config:            "testdata/networking/v1alpha3/service-entry-cloudflare.yaml",
+			url:               "https://www.cloudflare.com",
+			shouldBeReachable: true,
+			shouldBeTLS:       false,
+		},
+		{
+			name:              "REACHABLE_WITH_TLS_ORIGINATION_www.cloudflare.com_443_over_cloudflare_80",
```
Comment (hzxuzhonghu) [inline tests/e2e/tests/pilot/externalservice_tls_origination_test.go]: 80 over 443

File: tests/e2e/tests/pilot/externalservice_tls_origination_test.go (hunk 2 above)
Comment (hzxuzhonghu) [reply tests/e2e/tests/pilot/externalservice_tls_origination_test.go]: i know this is the current behavior, i am just saying user expected.

**Pull request closed without merging**